src/TestFramework/TestFramework/Assertions/CollectionAssert.cs

```diff
@@ -1444,5 +1625,12 @@ private class ObjectComparer : IComparer
         int IComparer.Compare(object? x, object? y) => Equals(x, y) ? 0 : -1;
     }
 
+    private class ObjectEqualityComparer : IEqualityComparer<object>
```

Comment:
It's probably better to use `EqualityComparer.Default` instead of this custom class.